src/wp-includes/formatting.php

```diff
@@ -2739,6 +2739,10 @@ function addslashes_gpc( $gpc ) {
  *
  * @since 2.0.0
  *
+ * @phpstan-template T
```

Comment:
I'm in two minds about this. These PHPStan-specific tags mean that developers who use these functions in plugins and themes get accurate type information when they use PHPStan on their own project, which is really valuable. On the other hand, if WordPress core doesn't implement PHPStan as part of its tooling then I can see that it could be confusing.

I see these sorts of annotations in the same way I see PHPCS/WPCS annotations, and indeed PHPDoc in general. If the developer reading them isn't familiar with the tooling then it'll always be confusing, it's just that we're more used to PHPDoc and PHPCS as it's more widely used.

My preference would be to retain them if we do decide to implement PHPStan tooling, but pull them back out if we don't